backend/app/routes.py

```diff
@@ -11,14 +11,32 @@
 
 @app.route('/')
 def index():
+    """Provide basic documentation about the available resources.
+    
+    All endpoints return JSON-formatted data.
+    """
     return jsonify({
-        'description': 'Travel Time App backend root',
-        'endpoints': [str(rule) for rule in app.url_map.iter_rules()]
+        'description': 'Travel Time App backend',
+        'available_endpoints': [
+            {
+                'path': str(rule),
+                'docstring': app.view_functions[rule.endpoint].__doc__
+            } for rule in app.url_map.iter_rules()
+        ]
     })
 
 # test URL /closest-node/-79.3400/43.6610

```

Comment:
I think this is fine for now. This is not really a requirement for now, but I wonder in the future whether we can sequester all test cases listed in this py file into a separate py file (with added formal unit testing)

Reply:
Absolutely. Great idea! 

Unit testing is one thing, but I've wondered about integration testing because so many of these functions just fire off SQL queries to the database. And how do we do automated testing that requires a connection to the database?